coffee-frontend/src/components/Login/signup/Signup.js

```diff
@@ -1,113 +1,131 @@
-                state: { from: this.props.location }
-            }}/>;            
-        }
-
-        return (
-            <div className="signup-container">
-                <div className="signup-content">
-                    <h1 className="signup-title">Signup with SpringSocial</h1>
-                    <SocialSignup />
-                    <div className="or-separator">
-                        <span className="or-text">OR</span>
-                    </div>
-                    <SignupForm {...this.props} />
-                    <span className="login-link">Already have an account? <Link to="/login">Login!</Link></span>
-                </div>
-            </div>
-        );
+const useStylesLogin = makeStyles((theme) =>
+  createStyles({
+    container: {
+      display: 'flex',
+      flexWrap: 'wrap',
+      margin: `${theme.spacing(0)} auto`
+    },
+    loginBtn: {
+      marginTop: theme.spacing(2),
+      flexGrow: 1
+    },
+    card: {
+      marginTop: theme.spacing(10),
+      width: '100%',
     }
-}
 
+  }),
+);
 
-class SocialSignup extends Component {
-    render() {
-        return (
-            <div className="social-signup">
-                <a className="btn btn-block social-btn google" href={GOOGLE_AUTH_URL}>
-                    <img src={googleLogo} alt="Google" /> Sign up with Google</a>
-                <a className="btn btn-block social-btn facebook" href={FACEBOOK_AUTH_URL}>
-                    <img src={fbLogo} alt="Facebook" /> Sign up with Facebook</a>
-            </div>
-        );
-    }
-}
 
-class SignupForm extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            name: '',
-            email: '',
-            password: ''
-        }
-        this.handleInputChange = this.handleInputChange.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
+function SignupForm() {
+    const classes = useStylesLogin();
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [name, setName] = useState('');
 
-    handleInputChange(event) {
-        const target = event.target;
-        const inputName = target.name;        
-        const inputValue = target.value;
+    const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+    const [helperText, setHelperText] = useState('');
+    const [error, setError] = useState(false);
+  
+    useEffect(() => {
+      if (email.trim() && password.trim()) {
+        setIsButtonDisabled(false);
+      } else {
+        setIsButtonDisabled(true);
+      }
+    }, [email, password]);
+  
+    const handleSignup = async () => {
+      if (password.length > 3) {
+        setError(false);
+        setHelperText('Login Successfully');
+        const signuprequest = {email, password, name}
+        const signupResponse = await signup(signuprequest)
+      } else {
+        setError(true);
+        setHelperText('Password is not secure enough')
```

Comment:
I suggest we give better hints to the user, more like: `user@example.com has already chosen that password. Please choose another, unique password`.

Reply:
What?

Reply:
;-))))